app/helpers/spree/admin/base_helper.rb

```diff
@@ -88,8 +86,6 @@ def preference_field_options(options)
             { size: 10, class: 'input_integer', step: :any }
           when :boolean
             {}
-          when :string
-            { size: 10, class: 'input_string fullwidth' }
```

Comment:
Don't know if this is a problem but I noticed that because merge! is used the attributes will always be overwritten by the argument hash. This means that the size attributes defined in the case hashes are never actually returned in the final field_options. If the options[:size] passed to the merge! method is nil, this will be the returned value. (see also the tests that were added)